### ISSUE TITLE
Why is data not shuffled in split_data.py?

### ISSUE THREAD

**Issue by xiahaoyun** [Kuaishou-OneRec/OpenOneRec#25]
I noticed that `data/scripts/split_data.py` doesn't perform a shuffle for different data types; it simply concatenates the data and outputs it as a file of 1000 lines each. I'd like to know why a global shuffle isn't performed before saving?

**Comment by zzx-zev** [comment]
Thanks for pointing this out. This is a good point.

In our internal pipeline, we developed a DataHub platform to manage data at the dataset level, where a dataset is organized into multiple files. Because the pre-training data is extremely massive, performing a global shuffle is time-consuming. Instead, we implement shuffling within the dataloader at both file level and sample level. (see [qwen3_dataset.py](https://github.com/Kuaishou-OneRec/OpenOneRec/blob/main/pretrain/onerec_llm/data/qwen3_dataset.py) for details)

This repo follows the same configuration. But given the current size of our open-source data, global shuffle may be feasible now. We'll submit a fix soon to support it. 

Thanks again for opening the issue!
